src/include/migraphx/shape.hpp

```diff
@@ -87,12 +89,12 @@ struct shape
     {
         std::size_t min = 0;
         std::size_t max = 0;
-        std::size_t opt = 0;
+        std::set<std::size_t> opts{};
```

Comment:
Can we call this `optimals`?

Reply:
I can make the change in a later PR. Making the change here will likely delay things. Added a TODO note.